src/handler.ts

```diff
@@ -0,0 +1,192 @@
+import * as assert from "assert";
+import { APIGatewayProxyHandler } from "aws-lambda";
+import { DynamoDB } from "aws-sdk";
+import * as uuid from "uuid/v4";
+import getConfig from "./getConfig";
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+
+type Container = {
+  config: {
+    DynamoDB: object;
+    stage: string;
+  };
+  dynamodb: DynamoDB;
+  client: DocumentClient;
+  TableName: string;
+};
+const retrieveContainer: () => Container = () => {
```

Comment:
The inversion of control pattern is something I like when dealing with endpoints as methods.

Reply:
As the implementation grows, it is likely that the `retrieveContainer` method will spend cpu cycles on things it may not necessarily need to fetch for the specific request.  The tradeoff for this is that an engineer working on an endpoint will generally not have to think about gathering any resources needed to implement a solution.

I have used IoC when working on an application deployed via docker.  After some thought, it is probably something to be avoided in lambda at scale.